src/transformers/generation_tf_utils.py

```diff
@@ -2484,9 +2484,6 @@ def beam_search(
 
         def flatten_beam_dim(tensor, batch_axis=0):
             """Flattens the first two dimensions of a non-scalar array."""
-            # ignore scalars (e.g. cache index)
-            if tf.rank(tensor) == 0:
-                return tensor
```

Comment:
This pattern was inherited from FLAX. Contrarily to FLAX, it seems, this `if` throws the XLA compiler of balance, causing `tensor` to have an unknown shape. All problems I was seeing before were a downstream consequence of that unknown shape.